google/cloud/testing_util/assert_ok.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_ASSERT_OK_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_ASSERT_OK_H_
+
+#include "google/cloud/status.h"
+#include <gtest/gtest.h>
+
+namespace testing {
+namespace internal {
+
+// A unary predicate-formatter for google::cloud::Status.
+testing::AssertionResult PredFormatStatus(
```

Comment:
It would be great if ASSERT_OK() (and the expect variant) worked for both Status and StatusOr. I'm guessing here, but it looks like you may be able to overload PredFormatStatus() for both Status and StatusOr, and handle each of them the correct way?

If it handles both, we may want to name it something a tad more generic, like maybe `IsOkPredicate` or something?

Reply:
Done, PTAL.